src/scaffold.rs

```diff
@@ -364,6 +416,54 @@ mod tests {
         )
     }
 
+    #[test]
+    fn build_objmetadata_when_no_annotation() {
+        let mut metadata = mock_metadata_with_no_annotations();
+        metadata.protocol_version = Some(policy_evaluator::ProtocolVersion::V1);
+        let scaffold_data = ScaffoldPolicyData {
+            uri: "not_relevant".to_string(),
+            policy_title: Some("test".to_string()),
+            metadata,
+            settings: Default::default(),
+        };
+
+        let obj_metadata = build_objmetadata(scaffold_data);
+        assert!(obj_metadata.annotations.is_none());
+    }
+
+    #[test]
+    fn build_objmetadata_with_annot_severity_category() {
+        let mut metadata = mock_metadata_with_severity_category();
+        metadata.protocol_version = Some(policy_evaluator::ProtocolVersion::V1);
+        let scaffold_data = ScaffoldPolicyData {
+            uri: "not_relevant".to_string(),
+            policy_title: Some("test".to_string()),
+            metadata,
+            settings: Default::default(),
+        };
+
+        let obj_metadata = build_objmetadata(scaffold_data);
+        assert!(obj_metadata.annotations.is_some());
+        assert_eq!(
+            obj_metadata
+                .annotations
+                .as_ref()
+                .unwrap()
+                .get(KUBEWARDEN_ANNOTATION_POLICY_SEVERITY)
+                .unwrap(),
+            &String::from("medium")
+        );
+        assert_eq!(
+            obj_metadata
+                .annotations
+                .as_ref()
+                .unwrap()
+                .get(KUBEWARDEN_ANNOTATION_POLICY_CATEGORY)
+                .unwrap(),
+            &String::from("PSP")
+        );
```

Comment:
Maybe you can use the [`expect`](https://doc.rust-lang.org/std/result/enum.Result.html#method.expect) method. Thus, you can validate the content and panic with a proper error message when it's invalid.

Reply:
refactored in https://github.com/kubewarden/kwctl/pull/532. I wonder if this is what you meant, or you meant substituting the whole `assert_eq!()`?